patches/tModLoader/Terraria/ModLoader/ModAchievement.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Xna.Framework.Graphics;
+using ReLogic.Content;
+using Terraria.Achievements;
+using Terraria.GameContent.Achievements;
+using Terraria.Localization;
+
+namespace Terraria.ModLoader;
+
+/// <summary>
+/// Define a custom achievement and implement how it should act upon completion
+/// </summary>
+public abstract class ModAchievement : ModType<Achievement, ModAchievement>, ILocalizedModType
+{
+    public Achievement Achievement => Entity;
+
+    /// <summary>
+    /// Achievement type, this is used for keeping track of modded achievements with ModContent.GetModAchievement(int type);
+    /// </summary>
+    public int Type => Achievement.Type;
+
+    private string TextureName => (GetType().Namespace + "." + Name).Replace('.', '/');//GetType().FullName.Replace('.', '/');
+    private Asset<Texture2D> Texture;
+    public string LocalizationCategory => "Achievements";
+
+    public override sealed bool IsCloneable => false;
+
+    /// <summary>
+    /// Gets the localized friendly name of the achievement.
+    /// </summary>
+    public virtual LocalizedText FriendlyName => Mod.GetLocalization($"{LocalizationCategory}.{Name}.Name");
+
+    public AchievementCondition GetCondition(string condition) => Achievement.GetCondition(condition);
```

Comment:
If we follow the convention of having conditions be public fields on the the `ModAchivement` subclasses, we don't need this method.